src/contacts/view/ContactViewer.js

```diff
@@ -129,7 +129,9 @@ export class ContactViewer {
 									insertBetween([
 										this.contact.company ? m("span.company", this.contact.company) : null,
 										this.contact.role ? m("span.title", this.contact.role) : null,
-										this._hasBirthday() ? m("span.birthday", this._formatBirthday()) : null
+										this._hasBirthday() && this.contact.birthdayIso
+											? m("span.birthday", memoized(this._formatBirthday)(neverNull(this.contact.birthdayIso)))
```

Comment:
That's not exatly how memoized works. You need to keep the result of `memoized` in a field so that it caches the return value internally. Right now you create a new memoized function each time and call it so there's no difference.

Also, passing `this._formatBirthday` like this will lose `this` so I would just make it freestanding below to avoid confusion.

Reply:
Actually i looked at the code just realized that memoized only has a single "cache" entry. So yes, that does not make any sense in multiple ways...

Reply:
Ah , yes, it's also an array, isn't it? hmm

Reply:
The parameter `birthdayIso`? Thats actually a string...

Reply:
no, wait, I thought insertBetween operates on multiple dates or something but it doesn't, why would `memoized` work? you should only have 0 or 1 birthday here?

Reply:
Well, i guess technically it would work, but then again we dont update this contact view when the contact changes anyway (which was what i was going for). So we can just save the formatted string upon component creation.

Reply:
Wait, but we do update the view? We create the new one? Yeah, then it's easiest to just save it, otherwise I woul do memoize.

Reply:
Yes, if there is an entity update we create a new viewer. But actually, i just saw that `this.contactAppellation` is saved on the Viewer so it doesn't make too much sense not to also save the Birthday.